learn/security/tenant_tokens.md

```diff
@@ -0,0 +1,299 @@
+
+Meilisearch only supports JWT tokens and `typ` must always be `JWT`.
+
+### Step 2: Preparing the token payload
+
+Once the header is ready, you must create a payload. In this case, payload means the data containing the token's API key prefix, search rules, and optional expiry date.
+
+First, you must extract the first eight characters from a valid API key with access to the search endpoint:
+
+```js
+const meiliSearchApikey = '[an_api_key]';
+const yourApiKeyPrefix = meiliSearchApiKey.slice(0,8);
+```
+
+`apiKey` must be a valid Meilisearch API key. A tenant token will have access to the same indexes as the API key you use when generating it. When generating a tenant token payload, you should only supply the first 8 characters of a key.
+
+Your payload must also contain a `searchRules` object. The following example would force all searches made with the token to only return results with a `user_id` of `1`:
+
+```js
+const yourSearchRules = {
+  "patient_medical_records": {
+    "filter": "user_id = 1"
+  }
+}
+```
+
+`searchRules` must be an object containing at least one search rule. A search rule must specify either an index name or a `*` wildcard. Search rules must also declare at least one filter. In this example, the filter makes it so a user only has access to their own records.
+
+With both these steps completed, you are ready to generate the token payload:
+
+```js
+const payload = {
+  "apiKeyPrefix": yourApiKeyPrefix,
+  "exp": 1641835850,
+  "searchRules": yourSearchRules
+}
+```
+
+You can also optionally set an expiry date for your token. Tokens are invalid once past their expiry date.
+
+Like the header, it is necessary to encode the payload:
+
+```js
+const base64Payload = base64Encode(payload);
+```
+
+You can read more about each of element of the tenant token payload in [this guide's final section](#tenant-token-payload-reference).
+
+### Step 3: Signing a token
+
+Tokens following the JWT standard must be signed. The signature is created by concatenating the encoded header and payload using a `.` as the separator, then encrypting the resulting string. **The encryption key must be the same you used when generating the payload's API key prefix.**
+
+```js
+const signature = HS256(base64Header + '.' + base64Payload, meiliSearchApiKey);
+```
+
+This example uses an HS256 encryption algorithm. **The encryption algorithm you use must be the same as the one indicated in the token's header.**
+
+The remaining step is putting it all together. Concatenate the header, payload, and signature using a `.` to separate each element of the token:
+
+```js
+const yourTenantToken = base64Header + '.' + base64Payload + '.' + signature;
+```
+
+### Using a tenant token without an SDK
+
+Your token is ready. Tenant tokens can seamlessly replace any requests made to the search endpoint:
+
+<CodeSamples id="tenant_token_guide_search_no_sdk_1" />
```

Comment:
Is this a cURL code sample?

Reply:
Yes. L#660 on `.code-samples.meilisearch.yaml`